tools/generator/dfg/stm32/stm.py

```diff
@@ -89,6 +89,71 @@ def getGpioRemapForModuleConfig(module, config):
         mmm['mapping'] = stm32f1_gpio_remap[module]['mapping'][int(config)]
     return mmm
 
+stm32f3_dma_remap = \
+{
+    'dma1ch1': {
+        'tim17_ch1': 'tim17_up',
+        'tim17_up':  {'position': 12, 'clear': 1, 'set': 0},
+    },
+    'dma1ch2': {
+        'adc2':      {'position': 72, 'clear': 3, 'set': 2},
+        'i2c_tx':    {'position': 70, 'clear': 3, 'set': 1},
+        'spi_rx':    {'position': 64, 'clear': 3, 'set': 0}, # also 'set': 3
+    },
+    'dma1ch3': {
+        'dac1_ch1':  'tim6_up',
+        'i2c_rx':    {'position': 68, 'clear': 3, 'set': 1},
+        'spi_tx':    {'position': 66, 'clear': 3, 'set': 0}, # also 'set': 3
+        'tim16_ch1': 'tim16_up',
+        'tim16_up':  {'position': 11, 'clear': 1, 'set': 0},
+        'tim6_up':   {'position': 13, 'clear': 1, 'set': 1},
+    },
+    'dma1ch4': {
+        'adc2':      {'position': 72, 'clear': 3, 'set': 3},
+        'dac1_ch2':  'tim7_up',
+        'i2c_tx':    {'position': 70, 'clear': 3, 'set': 2},
+        'spi_rx':    {'position': 64, 'clear': 3, 'set': 1},
+        'tim7_up':   {'position': 14, 'clear': 1, 'set': 1},
+    },
+    'dma1ch5': {
+        'dac2_ch1':  'tim18_up',
+        'i2c_rx':    {'position': 68, 'clear': 3, 'set': 2},
+        'spi_tx':    {'position': 66, 'clear': 3, 'set': 1},
+        'tim18_up':  {'position': 15, 'clear': 1, 'set': 1},
+    },
+    'dma1ch6': {
+        'i2c_tx':    {'position': 70, 'clear': 3, 'set': 0}, # also 'set': 3
+        'spi_rx':    {'position': 64, 'clear': 3, 'set': 2},
+    },
+    'dma1ch7': {
+        'i2c_rx':    {'position': 68, 'clear': 3, 'set': 0}, # also 'set': 3
+        'spi_tx':    {'position': 66, 'clear': 3, 'set': 2},
+        'tim17_ch1': 'tim17_up',
+        'tim17_up':  {'position': 12, 'clear': 1, 'set': 1},
+    },
+
+
+    'dma2ch1': {
+        'adc2':     [{'position': 8, 'clear': 1, 'set': 0}, {'position': 73, 'clear': 1, 'set': 0}],
```

Comment:
Right, so… ADC2 can be remapped entirely to DMA2 or DMA1 with bit 73:
<img width="411" alt="screen shot 2018-09-19 at 20 06 35" src="https://user-images.githubusercontent.com/163066/45772332-aa42bb00-bc47-11e8-9a6c-4e1477fb1393.png">
But this bit doesn't set the channel, nono! DMA 1 channel is selected with bit 72:
<img width="411" alt="screen shot 2018-09-19 at 20 06 35 copy" src="https://user-images.githubusercontent.com/163066/45772387-cd6d6a80-bc47-11e8-8723-78f6c2c97188.png">
And DMA2 channel is selected with bit 8, *which also remaps ADC4*.
<img width="607" alt="screen shot 2018-09-19 at 20 06 23" src="https://user-images.githubusercontent.com/163066/45772397-d65e3c00-bc47-11e8-8837-8e7dbce71505.png">

Good lord, ST, really?